web/components/templates/sessions/sessionId/Tree/TreeView.tsx

```diff
@@ -283,6 +352,92 @@ const TreeView: React.FC<TreeViewProps> = ({
     }
   };
 
+  const timelineData = (() => {
```

Comment:
I love the functional-ness of this, can we move this to a useMemo inside a helper function or something?